### PR TITLE
Add logging framework.

### PR DESCRIPTION
**Pull request by kyler-m**

### CODE REVIEW

File: config/dn-log-conf.conf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+* GLOBAL:
+   FORMAT               =  "%datetime{%Y-%M-%d %H:%m:%s.%g} [%level] %msg"
+   FILENAME             =  "logs/nn-log.log"
```
Comment (pelmers) [inline config/dn-log-conf.conf]: should it log to same file as namenode? maybe you mean dn-log.log

File: rice-namenode/main.cc [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
 
+// initialize the logging library (only do this once!)
+INITIALIZE_EASYLOGGINGPP
+
+#define LOG_CONFIG_FILE "nn-log-conf.conf"
+
 using namespace client_namenode_translator;
 
 int main(int argc, char* argv[]) {
+	el::Configurations conf(LOG_CONFIG_FILE);
```
Comment (pelmers) [inline rice-namenode/main.cc:19]: inconsistent tabs/space here

**Review by pelmers — COMMENTED** [review]
Top-level remarks:
1. remove utility/logging/.ds_store
2. could replace the prints in the ERROR_AND_RETURN and ERROR_AND_FALSE macros with logs to error? (if there is no error log it would be nice to have one, but going to same file as other output is also okay)

Otherwise I approve. r+ with those changes

**Comment by pelmers** [comment]
Don't do these if it takes more than 5 minutes, but I would consider squashing all the readme tweaks into one commit (`rebase -i develop`, and put `f` in front of all tweak commits after the first).

And changing vagrant@vagrant the author field of commits to yourself (`commit --amend --author "first last x@y.com --no-edit"`).

**Comment by pelmers** [comment]
adding to above, the rebase would remove the no-op self-merge commit

**Comment by kyler-m** [comment]
I was planning on just taking advantage of Github's merge-and-squash to condense all of my commits into one

**Comment by pelmers** [comment]
oh, perfect -- disregard

**Pull request merged**